engine/access/apiproxy/access_api_proxy_test.go

```diff
@@ -135,45 +138,57 @@ func TestNewFlowCachedAccessAPIProxy(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Prepare a proxy that fails due to the second connection being idle
-	l := flow.IdentityList{{Address: unittest.IPPort("11634")}, {Address: unittest.IPPort("11635")}}
-	c := FlowAccessAPIForwarder{}
-	c.Forwarder, err = forwarder.NewForwarder(l, time.Second, grpcutils.DefaultMaxMsgSize)
-
-	if err == nil {
-		t.Fatal(fmt.Errorf("should not start with one connection ready"))
-	}
-
 	// Bring up 2nd upstream server
 	charlie2, _, err := newFlowLite("tcp", unittest.IPPort("11635"), done)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	background := context.Background()
+	// create the factory
+	connectionFactory := new(connection.ConnectionFactoryImpl)
```

Comment:
why do we create this way? I think the preferred way would be to use a struct initialization unless there is a strong reasoning